src/python/src/grpc/_adapter/_call.c

```diff
@@ -160,8 +160,22 @@ static const PyObject *pygrpc_call_accept(Call *self, PyObject *args) {
   return result;
 }
 
+static const PyObject *pygrpc_call_add_metadata(Call *self, PyObject *args) {
+  const char* key = NULL;
+  const char* value = NULL;
+  int value_length = 0;
+  if (!PyArg_ParseTuple(args, "ss#", &key, &value, &value_length)) {
+    return NULL;
+  }
+  grpc_metadata metadata;
+  metadata.key = key;
+  metadata.value = value;
+  metadata.value_length = value_length;
+  return pygrpc_translate_call_error(
+      grpc_call_add_metadata_old(self->c_call, &metadata, 0));
+}
+
 static const PyObject *pygrpc_call_premetadata(Call *self) {
```

Comment:
The only obvious naming schemes that come to mind end up also changing the names of all the other functions to have 'client' and 'server' variants, s.t. it becomes clearer that sending the initial metadata is explicitly a server-only thing by not having a client variant. Alternatively, more literally wrapping the C API s.t. the names on the Python type reflect those of the C type.

But, really, I'm ambivalent; this is all getting hidden from the public API anyway, and clarifying information would probably be better served in the doc string without making larger scale changes to nomenclature.